packages/ui/src/ui/pages/odin/controls/OdinOverview.tsx

```diff
@@ -229,67 +267,196 @@ function OdinOverview(props: OdinOverviewProps) {
 
     const dispatch = useDispatch();
 
+    const isTimeCloseToCurrent = (time: number) => moment().diff(moment(time), 'minutes') < 5;
+
+    const isPickerTimeValid = (time: number) => {
+        return time < new Date().getTime() - HALF_HOUR;
+    };
+
+    const onPrevDateClick = () => {
```

Comment:
It is better to move all the login to actions and leave here simple callbacks like:
```ts
const onPrevDateClick = () => {
    dispatch(fetchPrevOdinOverview(cluster));
}

const onNextDateClick = () => {
    dispatch(fetchNextOdinOverview(cluster));
}

const onUpdate = (value) => {
    dispatch(setOdinOverivewFromTime(value));
}
```

Reply:
good, fixed